src/main/java/org/eclipse/golo/compiler/ParseTreeToGoloIrVisitor.java

```diff
@@ -120,7 +89,21 @@ public void leaveNamedAugmentation() {
     }
 
     public void addFunction(GoloFunction function) {
-      this.functionContainersStack.peek().addFunction(function);
+      FunctionContainer container = this.functionContainersStack.peek();
+      if(container.getFunctions().contains(function)) {
+        GoloFunction firstDeclaration = null;
+        for (GoloFunction f : container.getFunctions()) {
+          if (function.equals(f)) {
+            firstDeclaration = f;
+          }
+        }
+        errorMessage(AMBIGUOUS_DECLARATION, function.getASTNode(),
+            String.format("Declaring a function `%s` twice (declared first at line=%d, column=%d) ",
+                function.getName(),
+                firstDeclaration.getASTNode().getLineInSourceCode(),
```

Comment:
You can directly use `GoloASTNode::getPositionInSourceCode` which override `toString`...